[MergeDuplicates] The Combine crashes when merging senses

In Merge Duplicates, drop one sense card into another. The following error occurs:

> Unexpected Application Error!
Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at rc (https://thecombine.app/static/js/main.8a50a610.js:2:630141)
    at Mi (https://thecombine.app/static/js/main.8a50a610.js:2:592925)
    at Object.callback (https://thecombine.app/static/js/main.8a50a610.js:2:592784)
    at https://thecombine.app/static/js/main.8a50a610.js:2:661880
    at uc (https://thecombine.app/static/js/main.8a50a610.js:2:633475)
    at Object.notify (https://thecombine.app/static/js/main.8a50a610.js:2:661854)
    at Object.notifyNestedSubs (https://thecombine.app/static/js/main.8a50a610.js:2:662343)
    at i (https://thecombine.app/static/js/main.8a50a610.js:2:661698)
    at v (https://thecombine.app/static/js/main.8a50a610.js:2:697534)
    at https://thecombine.app/static/js/main.8a50a610.js:2:423723

Here's a more detailed error in dev:
> react-dom.development.js:27292 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at checkForNestedUpdates (react-dom.development.js:27292:1)
    at scheduleUpdateOnFiber (react-dom.development.js:25475:1)
    at dispatchSetState (react-dom.development.js:17527:1)
    at setRef (setRef.js:16:1)
    at useForkRef.js:17:1
    at Array.forEach (<anonymous>)
    at useForkRef.js:16:1
    at safelyDetachRef (react-dom.development.js:22908:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24003:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1)
checkForNestedUpdates @ react-dom.development.js:27292
scheduleUpdateOnFiber @ react-dom.development.js:25475
dispatchSetState @ react-dom.development.js:17527
setRef @ setRef.js:16
(anonymous) @ useForkRef.js:17
(anonymous) @ useForkRef.js:16
safelyDetachRef @ react-dom.development.js:22908
commitDeletionEffectsOnFiber @ react-dom.development.js:24003
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
[...]
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24293
Show 199 more frames
Show less
console.js:213 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
    at PartOfSpeech (http://localhost:3000/static/js/bundle.js:12175:68)
    at div
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at CardContent (http://localhost:3000/static/js/bundle.js:28450:82)
    at SenseCardContent (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsCompleted_tsx.chunk.js:365:37)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Paper (http://localhost:3000/static/js/bundle.js:37780:83)
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Card (http://localhost:3000/static/js/bundle.js:28584:82)
    at Draggable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7069:58)
    at ConnectFunction (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:8235:68)
    at PrivateDraggable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7371:26)
    at PublicDraggable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7379:32)
    at DragSense (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:51:94)
    at div
    at div
    at Droppable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7390:69)
    at ConnectFunction (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:8235:68)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Paper (http://localhost:3000/static/js/bundle.js:37780:83)
    at DropWord (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:205:79)
    at li
    at http://localhost:3000/static/js/bundle.js:21976:66
    at ImageListItem (http://localhost:3000/static/js/vendors-node_modules_microsoft_signalr_dist_esm_HubConnectionBuilder_js-node_modules_mui_icon-a71297.chunk.js:17170:82)
    at ul
    at http://localhost:3000/static/js/bundle.js:21976:66
    at ImageList (http://localhost:3000/static/js/vendors-node_modules_microsoft_signalr_dist_esm_HubConnectionBuilder_js-node_modules_mui_icon-a71297.chunk.js:17402:82)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Grid (http://localhost:3000/static/js/bundle.js:32020:87)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Grid (http://localhost:3000/static/js/bundle.js:32020:87)
    at Provider (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7977:20)
    at App (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:6164:25)
    at ErrorBoundary (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:377:35)
    at DragDropContext (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:6288:19)
    at MergeDragDrop (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:705:79)
    at div
    at MergeDupsStep (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:1170:80)
    at MergeDups (http://localhost:3000/static/js/src_goals_MergeDuplicates_index_tsx.chunk.js:25:16)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at BaseGoalScreen (http://localhost:3000/static/js/src_goals_DefaultGoal_BaseGoalScreen_tsx.chunk.js:192:75)
    at LoadingGoalScreen (http://localhost:3000/static/js/src_goals_DefaultGoal_BaseGoalScreen_tsx.chunk.js:171:81)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at RenderedRoute (http://localhost:3000/static/js/bundle.js:100689:5)
    at Routes (http://localhost:3000/static/js/bundle.js:101311:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at AppWithBar (http://localhost:3000/static/js/src_components_App_AppLoggedIn_tsx.chunk.js:1298:76)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at RequireAuth (http://localhost:3000/static/js/bundle.js:17116:76)
    at RenderedRoute (http://localhost:3000/static/js/bundle.js:100689:5)
    at RenderErrorBoundary (http://localhost:3000/static/js/bundle.js:100636:5)
    at DataRoutes (http://localhost:3000/static/js/bundle.js:99299:5)
    at Router (http://localhost:3000/static/js/bundle.js:101249:15)
    at RouterProvider (http://localhost:3000/static/js/bundle.js:99141:5)
    at Suspense
    at div
    at App
    at PersistGate (http://localhost:3000/static/js/bundle.js:107759:5)
    at Provider (http://localhost:3000/static/js/bundle.js:96255:3)
    at SnackbarProvider (http://localhost:3000/static/js/bundle.js:68629:24)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at StyledEngineProvider (http://localhost:3000/static/js/bundle.js:44229:5)
overrideMethod @ console.js:213
printWarning @ react-dom.development.js:86
error @ react-dom.development.js:60
checkForNestedUpdates @ react-dom.development.js:27300
scheduleUpdateOnFiber @ react-dom.development.js:25475
dispatchSetState @ react-dom.development.js:17527
setRef @ setRef.js:16
(anonymous) @ useForkRef.js:17
(anonymous) @ useForkRef.js:16
safelyDetachRef @ react-dom.development.js:22908
commitDeletionEffectsOnFiber @ react-dom.development.js:24003
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
commitDeletionEffects @ react-dom.development.js:23976
recursivelyTraverseMutationEffects @ react-dom.development.js:24259
commitMutationEffectsOnFiber @ react-dom.development.js:24293
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24346
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24346
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24293
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24293
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24293
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24346
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24293
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
[...]
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
Show 200 more frames
Show less
console.js:213 The above error occurred in the <span> component:
    at span
    at Tooltip (http://localhost:3000/static/js/bundle.js:40851:83)
    at IconButtonWithTooltip (http://localhost:3000/static/js/bundle.js:11865:68)
    at CatGroupButton
    at PartOfSpeech (http://localhost:3000/static/js/bundle.js:12175:68)
    at div
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at CardContent (http://localhost:3000/static/js/bundle.js:28450:82)
    at SenseCardContent (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsCompleted_tsx.chunk.js:365:37)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Paper (http://localhost:3000/static/js/bundle.js:37780:83)
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Card (http://localhost:3000/static/js/bundle.js:28584:82)
    at Draggable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7069:58)
    at ConnectFunction (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:8235:68)
    at PrivateDraggable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7371:26)
    at PublicDraggable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7379:32)
    at DragSense (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:51:94)
    at div
    at div
    at Droppable (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7390:69)
    at ConnectFunction (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:8235:68)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Paper (http://localhost:3000/static/js/bundle.js:37780:83)
    at DropWord (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:205:79)
    at li
    at http://localhost:3000/static/js/bundle.js:21976:66
    at ImageListItem (http://localhost:3000/static/js/vendors-node_modules_microsoft_signalr_dist_esm_HubConnectionBuilder_js-node_modules_mui_icon-a71297.chunk.js:17170:82)
    at ul
    at http://localhost:3000/static/js/bundle.js:21976:66
    at ImageList (http://localhost:3000/static/js/vendors-node_modules_microsoft_signalr_dist_esm_HubConnectionBuilder_js-node_modules_mui_icon-a71297.chunk.js:17402:82)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Grid (http://localhost:3000/static/js/bundle.js:32020:87)
    at div
    at http://localhost:3000/static/js/bundle.js:21976:66
    at Grid (http://localhost:3000/static/js/bundle.js:32020:87)
    at Provider (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:7977:20)
    at App (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:6164:25)
    at ErrorBoundary (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:377:35)
    at DragDropContext (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:6288:19)
    at MergeDragDrop (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:705:79)
    at div
    at MergeDupsStep (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:1170:80)
    at MergeDups (http://localhost:3000/static/js/src_goals_MergeDuplicates_index_tsx.chunk.js:25:16)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at BaseGoalScreen (http://localhost:3000/static/js/src_goals_DefaultGoal_BaseGoalScreen_tsx.chunk.js:192:75)
    at LoadingGoalScreen (http://localhost:3000/static/js/src_goals_DefaultGoal_BaseGoalScreen_tsx.chunk.js:171:81)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at RenderedRoute (http://localhost:3000/static/js/bundle.js:100689:5)
    at Routes (http://localhost:3000/static/js/bundle.js:101311:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at AppWithBar (http://localhost:3000/static/js/src_components_App_AppLoggedIn_tsx.chunk.js:1298:76)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at RequireAuth (http://localhost:3000/static/js/bundle.js:17116:76)
    at RenderedRoute (http://localhost:3000/static/js/bundle.js:100689:5)
    at RenderErrorBoundary (http://localhost:3000/static/js/bundle.js:100636:5)
    at DataRoutes (http://localhost:3000/static/js/bundle.js:99299:5)
    at Router (http://localhost:3000/static/js/bundle.js:101249:15)
    at RouterProvider (http://localhost:3000/static/js/bundle.js:99141:5)
    at Suspense
    at div
    at App
    at PersistGate (http://localhost:3000/static/js/bundle.js:107759:5)
    at Provider (http://localhost:3000/static/js/bundle.js:96255:3)
    at SnackbarProvider (http://localhost:3000/static/js/bundle.js:68629:24)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at StyledEngineProvider (http://localhost:3000/static/js/bundle.js:44229:5)
React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.
overrideMethod @ console.js:213
logCapturedError @ react-dom.development.js:18687
callback @ react-dom.development.js:18755
callCallback @ react-dom.development.js:13923
commitUpdateQueue @ react-dom.development.js:13944
commitLayoutEffectOnFiber @ react-dom.development.js:23364
commitLayoutMountEffects_complete @ react-dom.development.js:24688
commitLayoutEffects_begin @ react-dom.development.js:24674
commitLayoutEffects_begin @ react-dom.development.js:24656
commitLayoutEffects @ react-dom.development.js:24612
commitRootImpl @ react-dom.development.js:26823
commitRoot @ react-dom.development.js:26682
performSyncWorkOnRoot @ react-dom.development.js:26117
flushSyncCallbacks @ react-dom.development.js:12042
(anonymous) @ react-dom.development.js:25651
Show 15 more frames
Show less
react-dom.development.js:27292 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at checkForNestedUpdates (react-dom.development.js:27292:1)
    at scheduleUpdateOnFiber (react-dom.development.js:25475:1)
    at dispatchSetState (react-dom.development.js:17527:1)
    at setRef (setRef.js:16:1)
    at useForkRef.js:17:1
    at Array.forEach (<anonymous>)
    at useForkRef.js:16:1
    at safelyDetachRef (react-dom.development.js:22908:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24003:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1)
checkForNestedUpdates @ react-dom.development.js:27292
scheduleUpdateOnFiber @ react-dom.development.js:25475
dispatchSetState @ react-dom.development.js:17527
setRef @ setRef.js:16
(anonymous) @ useForkRef.js:17
(anonymous) @ useForkRef.js:16
safelyDetachRef @ react-dom.development.js:22908
commitDeletionEffectsOnFiber @ react-dom.development.js:24003
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
recursivelyTraverseDeletionEffects @ react-dom.development.js:23989
commitDeletionEffectsOnFiber @ react-dom.development.js:24118
commitDeletionEffects @ react-dom.development.js:23976
recursivelyTraverseMutationEffects @ react-dom.development.js:24259
commitMutationEffectsOnFiber @ react-dom.development.js:24293
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24346
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24346
[...]
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24576
recursivelyTraverseMutationEffects @ react-dom.development.js:24273
commitMutationEffectsOnFiber @ react-dom.development.js:24293
Show 199 more frames
Show less
console.js:213 Error handled by React Router default ErrorBoundary: Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at checkForNestedUpdates (react-dom.development.js:27292:1)
    at scheduleUpdateOnFiber (react-dom.development.js:25475:1)
    at dispatchSetState (react-dom.development.js:17527:1)
    at setRef (setRef.js:16:1)
    at useForkRef.js:17:1
    at Array.forEach (<anonymous>)
    at useForkRef.js:16:1
    at safelyDetachRef (react-dom.development.js:22908:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24003:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1) 
    at DefaultErrorComponent (http://localhost:3000/static/js/bundle.js:100604:15)
    at RenderErrorBoundary (http://localhost:3000/static/js/bundle.js:100636:5)
    at DataRoutes (http://localhost:3000/static/js/bundle.js:99299:5)
    at Router (http://localhost:3000/static/js/bundle.js:101249:15)
    at RouterProvider (http://localhost:3000/static/js/bundle.js:99141:5)
    at Suspense
    at div
    at App
    at PersistGate (http://localhost:3000/static/js/bundle.js:107759:5)
    at Provider (http://localhost:3000/static/js/bundle.js:96255:3)
    at SnackbarProvider (http://localhost:3000/static/js/bundle.js:68629:24)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at StyledEngineProvider (http://localhost:3000/static/js/bundle.js:44229:5)
overrideMethod @ console.js:213
DefaultErrorComponent @ hooks.tsx:515
renderWithHooks @ react-dom.development.js:16305
mountIndeterminateComponent @ react-dom.development.js:20074
beginWork @ react-dom.development.js:21587
beginWork$1 @ react-dom.development.js:27426
performUnitOfWork @ react-dom.development.js:26557
workLoopSync @ react-dom.development.js:26466
renderRootSync @ react-dom.development.js:26434
performSyncWorkOnRoot @ react-dom.development.js:26085
flushSyncCallbacks @ react-dom.development.js:12042
(anonymous) @ react-dom.development.js:25651
Show 12 more frames
Show less
console.js:213 The above error occurred in the <ErrorBoundary> component:
    at ErrorBoundary (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:377:35)
    at DragDropContext (http://localhost:3000/static/js/vendors-node_modules_mui_icons-material_esm_ArrowForwardIos_js-node_modules_mui_icons-materia-b1f74a.chunk.js:6288:19)
    at MergeDragDrop (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:705:79)
    at div
    at MergeDupsStep (http://localhost:3000/static/js/src_goals_MergeDuplicates_MergeDupsStep_index_tsx.chunk.js:1170:80)
    at MergeDups (http://localhost:3000/static/js/src_goals_MergeDuplicates_index_tsx.chunk.js:25:16)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at BaseGoalScreen (http://localhost:3000/static/js/src_goals_DefaultGoal_BaseGoalScreen_tsx.chunk.js:192:75)
    at LoadingGoalScreen (http://localhost:3000/static/js/src_goals_DefaultGoal_BaseGoalScreen_tsx.chunk.js:171:81)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at RenderedRoute (http://localhost:3000/static/js/bundle.js:100689:5)
    at Routes (http://localhost:3000/static/js/bundle.js:101311:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at AppWithBar (http://localhost:3000/static/js/src_components_App_AppLoggedIn_tsx.chunk.js:1298:76)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:23549:34)
    at LoadableWithChunkExtractor
    at Loadable
    at RequireAuth (http://localhost:3000/static/js/bundle.js:17116:76)
    at RenderedRoute (http://localhost:3000/static/js/bundle.js:100689:5)
    at RenderErrorBoundary (http://localhost:3000/static/js/bundle.js:100636:5)
    at DataRoutes (http://localhost:3000/static/js/bundle.js:99299:5)
    at Router (http://localhost:3000/static/js/bundle.js:101249:15)
    at RouterProvider (http://localhost:3000/static/js/bundle.js:99141:5)
    at Suspense
    at div
    at App
    at PersistGate (http://localhost:3000/static/js/bundle.js:107759:5)
    at Provider (http://localhost:3000/static/js/bundle.js:96255:3)
    at SnackbarProvider (http://localhost:3000/static/js/bundle.js:68629:24)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at StyledEngineProvider (http://localhost:3000/static/js/bundle.js:44229:5)
React will try to recreate this component tree from scratch using the error boundary you provided, RenderErrorBoundary.
overrideMethod @ console.js:213
logCapturedError @ react-dom.development.js:18687
callback @ react-dom.development.js:18755
callCallback @ react-dom.development.js:13923
commitUpdateQueue @ react-dom.development.js:13944
commitLayoutEffectOnFiber @ react-dom.development.js:23364
commitLayoutMountEffects_complete @ react-dom.development.js:24688
commitLayoutEffects_begin @ react-dom.development.js:24674
commitLayoutEffects_begin @ react-dom.development.js:24656
commitLayoutEffects @ react-dom.development.js:24612
commitRootImpl @ react-dom.development.js:26823
commitRoot @ react-dom.development.js:26682
performSyncWorkOnRoot @ react-dom.development.js:26117
flushSyncCallbacks @ react-dom.development.js:12042
(anonymous) @ react-dom.development.js:25651
Show 15 more frames
Show less
console.js:213 React Router caught the following error during render Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at checkForNestedUpdates (react-dom.development.js:27292:1)
    at scheduleUpdateOnFiber (react-dom.development.js:25475:1)
    at dispatchSetState (react-dom.development.js:17527:1)
    at setRef (setRef.js:16:1)
    at useForkRef.js:17:1
    at Array.forEach (<anonymous>)
    at useForkRef.js:16:1
    at safelyDetachRef (react-dom.development.js:22908:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24003:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1) {componentStack: '\n    at ErrorBoundary (http://localhost:3000/stati…ttp://localhost:3000/static/js/bundle.js:44229:5)'} 
    at RenderErrorBoundary (http://localhost:3000/static/js/bundle.js:100636:5)
    at DataRoutes (http://localhost:3000/static/js/bundle.js:99299:5)
    at Router (http://localhost:3000/static/js/bundle.js:101249:15)
    at RouterProvider (http://localhost:3000/static/js/bundle.js:99141:5)
    at Suspense
    at div
    at App
    at PersistGate (http://localhost:3000/static/js/bundle.js:107759:5)
    at Provider (http://localhost:3000/static/js/bundle.js:96255:3)
    at SnackbarProvider (http://localhost:3000/static/js/bundle.js:68629:24)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44046:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:44436:5)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:42252:14)
    at StyledEngineProvider (http://localhost:3000/static/js/bundle.js:44229:5)

--- a/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/DropWord.tsx
+++ b/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/DropWord.tsx
@@ -11,22 +11,25 @@ import {
   flagWord,
   setVern,
 } from "goals/MergeDuplicates/Redux/MergeDupsActions";
-import { MergeTreeState } from "goals/MergeDuplicates/Redux/MergeDupsReduxTypes";
-import { useAppDispatch } from "types/hooks";
+import { StoreState } from "types";
+import { useAppDispatch, useAppSelector } from "types/hooks";
 import theme from "types/theme";
 import { TypographyWithFont } from "utilities/fontComponents";
 
 interface DropWordProps {
-  mergeState: MergeTreeState;
   wordId: string;
 }
 
 export default function DropWord(props: DropWordProps): ReactElement {
   const dispatch = useAppDispatch();
+  const data = useAppSelector(
+    (state: StoreState) => state.mergeDuplicateGoal.data
+  );
+  const treeWord = useAppSelector(
+    (state: StoreState) => state.mergeDuplicateGoal.tree.words[props.wordId]
+  );
   const { t } = useTranslation();
 
-  const treeWord = props.mergeState.tree.words[props.wordId];
-  const data = props.mergeState.data;
   let protectedWithOneChild = false;
   const verns: string[] = [];
   if (treeWord) {

--- a/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/SidebarDragSense.tsx
+++ b/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/SidebarDragSense.tsx
@@ -7,12 +7,12 @@ import SenseCardContent from "goals/MergeDuplicates/MergeDupsStep/SenseCardConte
 import {
   MergeTreeReference,
   MergeTreeSense,
-  Sidebar,
 } from "goals/MergeDuplicates/MergeDupsTreeTypes";
+import { StoreState } from "types";
+import { useAppSelector } from "types/hooks";
 import theme from "types/theme";
 
 interface SidebarDragSenseProps {
-  sidebar: Sidebar;
   sense: MergeTreeSense;
   index: number;
 }
@@ -20,16 +20,17 @@ interface SidebarDragSenseProps {
 export default function SidebarDragSense(
   props: SidebarDragSenseProps
 ): ReactElement {
-  const ref: MergeTreeReference = {
-    wordId: props.sidebar.wordId,
-    mergeSenseId: props.sidebar.mergeSenseId,
-    order: props.index,
-  };
+  const draggableId = useAppSelector((state: StoreState) => {
+    const { mergeSenseId, wordId } = state.mergeDuplicateGoal.tree.sidebar;
+    const order = props.index;
+    const ref: MergeTreeReference = { wordId, mergeSenseId, order };
+    return JSON.stringify(ref);
+  });
 
   return (
     <Draggable
       key={props.sense.guid}
-      draggableId={JSON.stringify(ref)}
+      draggableId={draggableId}
       index={props.index}
       isDragDisabled={props.sense.protected}
     >

--- a/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/SidebarDrop.tsx
+++ b/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/SidebarDrop.tsx
@@ -4,25 +4,25 @@ import { ReactElement } from "react";
 import { Droppable } from "react-beautiful-dnd";
 
 import SidebarDragSense from "goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/SidebarDragSense";
-import {
-  MergeTreeSense,
-  Sidebar,
-} from "goals/MergeDuplicates/MergeDupsTreeTypes";
+import { MergeTreeSense } from "goals/MergeDuplicates/MergeDupsTreeTypes";
 import { setSidebar } from "goals/MergeDuplicates/Redux/MergeDupsActions";
-import { useAppDispatch } from "types/hooks";
+import { StoreState } from "types";
+import { useAppDispatch, useAppSelector } from "types/hooks";
 
-interface SidebarDropProps {
-  sidebar: Sidebar;
-  vernacular: string;
-}
-
-export default function SidebarDrop(props: SidebarDropProps): ReactElement {
+export default function SidebarDrop(): ReactElement {
   const dispatch = useAppDispatch();
+  const sidebar = useAppSelector(
+    (state: StoreState) => state.mergeDuplicateGoal.tree.sidebar
+  );
+  const vernacular = useAppSelector((state: StoreState) => {
+    const tree = state.mergeDuplicateGoal.tree;
+    return tree.words[tree.sidebar.wordId]?.vern;
+  });
 
   return (
     <Droppable
-      droppableId={`${props.sidebar.wordId} ${props.sidebar.mergeSenseId}`}
-      key={props.sidebar.mergeSenseId}
+      droppableId={`${sidebar.wordId} ${sidebar.mergeSenseId}`}
+      key={sidebar.mergeSenseId}
     >
       {(providedDroppable): ReactElement => (
         <div
@@ -37,14 +37,9 @@ export default function SidebarDrop(props: SidebarDropProps): ReactElement {
           >
             <ArrowForwardIos />
           </IconButton>
-          <Typography variant="h5">{props.vernacular}</Typography>
-          {props.sidebar.senses.map((sense: MergeTreeSense, index: number) => (
-            <SidebarDragSense
-              key={index}
-              index={index}
-              sidebar={props.sidebar}
-              sense={sense}
-            />
+          <Typography variant="h5">{vernacular}</Typography>
+          {sidebar.senses.map((sense: MergeTreeSense, index: number) => (
+            <SidebarDragSense key={index} index={index} sense={sense} />
           ))}
           {providedDroppable.placeholder}
         </div>

--- a/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/index.tsx
+++ b/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/index.tsx
@@ -24,11 +24,11 @@ export const trashId = "trash-drop";
 
 export default function MergeDragDrop(): ReactElement {
   const dispatch = useAppDispatch();
-  const sidebarHasMultipleSenses = useAppSelector(
+  const sidebarOpen = useAppSelector(
     (state: StoreState) =>
       state.mergeDuplicateGoal.tree.sidebar.senses.length > 1
   );
-  const sidebarSensesProtected = useAppSelector((state: StoreState) => {
+  const sidebarProtected = useAppSelector((state: StoreState) => {
     const senses = state.mergeDuplicateGoal.tree.sidebar.senses;
     return senses.length && senses[0].protected;
   });
@@ -95,7 +95,7 @@ export default function MergeDragDrop(): ReactElement {
         const destOrder = res.destination.index;
         if (
           src.order === destOrder ||
-          (destOrder === 0 && src.order !== undefined && sidebarSensesProtected)
+          (destOrder === 0 && src.order !== undefined && sidebarProtected)
         ) {
           // If the sense wasn't moved or was moved within the sidebar above a protected sense, do nothing.
           return;
@@ -111,14 +111,11 @@ export default function MergeDragDrop(): ReactElement {
   }
 
   function renderSidebar(): ReactElement {
-    if (!sidebarHasMultipleSenses) {
-      return <div />;
-    }
-    return (
+    return sidebarOpen ? (
       <Drawer
         anchor="right"
         variant="persistent"
-        open={sidebarHasMultipleSenses}
+        open={sidebarOpen}
         SlideProps={{
           style: {
             height: `calc(100% - ${appBarHeight}px)`,
@@ -128,6 +125,8 @@ export default function MergeDragDrop(): ReactElement {
       >
         <SidebarDrop />
       </Drawer>
+    ) : (
+      <div />
     );
   }
 

--- a/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/index.tsx
+++ b/src/goals/MergeDuplicates/MergeDupsStep/MergeDragDrop/index.tsx
@@ -24,19 +24,25 @@ export const trashId = "trash-drop";
 
 export default function MergeDragDrop(): ReactElement {
   const dispatch = useAppDispatch();
-  const mergeState = useAppSelector(
-    (state: StoreState) => state.mergeDuplicateGoal
+  const sidebarHasMultipleSenses = useAppSelector(
+    (state: StoreState) =>
+      state.mergeDuplicateGoal.tree.sidebar.senses.length > 1
   );
+  const sidebarSensesProtected = useAppSelector((state: StoreState) => {
+    const senses = state.mergeDuplicateGoal.tree.sidebar.senses;
+    return senses.length && senses[0].protected;
+  });
+  const words = useAppSelector(
+    (state: StoreState) => state.mergeDuplicateGoal.tree.words
+  );
+
   const [senseToDelete, setSenseToDelete] = useState<string>("");
   const { t } = useTranslation();
-
-  const sidebar = mergeState.tree.sidebar;
-  const treeWords = mergeState.tree.words;
 
   function handleDrop(res: DropResult): void {
     const src: MergeTreeReference = JSON.parse(res.draggableId);
     const srcWordId = res.source.droppableId;
-    const srcWord = treeWords[srcWordId];
+    const srcWord = words[srcWordId];
     if (srcWord?.protected && Object.keys(srcWord.sensesGuids).length === 1) {
       // Case 0: The final sense of a protected word cannot be moved.
       return;
@@ -89,9 +95,7 @@ export default function MergeDragDrop(): ReactElement {
         const destOrder = res.destination.index;
         if (
           src.order === destOrder ||
-          (destOrder === 0 &&
-            src.order !== undefined &&
-            sidebar.senses[0].protected)
+          (destOrder === 0 && src.order !== undefined && sidebarSensesProtected)
         ) {
           // If the sense wasn't moved or was moved within the sidebar above a protected sense, do nothing.
           return;
@@ -107,14 +111,14 @@ export default function MergeDragDrop(): ReactElement {
   }
 
   function renderSidebar(): ReactElement {
-    if (sidebar.senses.length <= 1) {
+    if (!sidebarHasMultipleSenses) {
       return <div />;
     }
     return (
       <Drawer
         anchor="right"
         variant="persistent"
-        open={sidebar.senses.length > 1}
+        open={sidebarHasMultipleSenses}
         SlideProps={{
           style: {
             height: `calc(100% - ${appBarHeight}px)`,
@@ -122,16 +126,13 @@ export default function MergeDragDrop(): ReactElement {
           },
         }}
       >
-        <SidebarDrop
-          sidebar={sidebar}
-          vernacular={treeWords[sidebar.wordId]?.vern}
-        />
+        <SidebarDrop />
       </Drawer>
     );
   }
 
   const newId = v4();
-  const colCount = Object.keys(treeWords).length + 1; // +1 for extra empty word.
+  const colCount = Object.keys(words).length + 1; // +1 for extra empty word.
 
   // This prevents things from moving when a draggable is dragged over the trash.
   const trashPlaceholderStyle: CSSProperties = {
@@ -157,16 +158,16 @@ export default function MergeDragDrop(): ReactElement {
         </Grid>
         <Grid item sm={11} xs={10 /* Allow trash icon more space. */}>
           <ImageList rowHeight="auto" cols={colCount} style={{ width: "90vw" }}>
-            {Object.keys(treeWords).map((key) => (
+            {Object.keys(words).map((key) => (
               <ImageListItem
                 key={key}
                 style={{ height: "70vh", margin: theme.spacing(1) }}
               >
-                <DropWord mergeState={mergeState} wordId={key} />
+                <DropWord wordId={key} />
               </ImageListItem>
             ))}
             <ImageListItem key={newId} style={{ margin: theme.spacing(1) }}>
-              <DropWord mergeState={mergeState} wordId={newId} />
+              <DropWord wordId={newId} />
             </ImageListItem>
             {renderSidebar()}
             <CancelConfirmDialog

